features/375-static-cdn/design.md

```diff
@@ -0,0 +1,138 @@
+# Static Asset Hosting
+
+## Background
+Developer would want to host static content (e.g. single-page-application files) on Skygear and serve them through
+high performance CDN.
+
+## Objectives
+- Developers should be able to upload static content to be hosted at specific path.
+- Hosted static content should be served & cached with CDN if possible.
+- SPA style routing should be supported.
+
+## Design
+
+### Static Deployment
+
+A new type of deployment, `static`, would be supported. This deployment type
+supports configurating a deployment context (`context`) and fallback absolute
+path (`fallback`).
+
+When `static` deployment type is used:
+- The deployment context would be persisted to object storage.
```

Comment:
We need to specify how do we handle symlinks. Some options:

- Silently ignore them (not recommended) 
- Error on encountering them
- Warn on encountering them
- Follow them
- Research on how popular object storage CLI handle them and follow their approach

Reply:
I would prefer to follow the symlink at client side before putting it into archive. It seems S3 CLI do the same thing (follow symlink) by default.

Reply:
Great. On implementation side, we need to detect circular symlinks.

Reply:
I think we can just leave it to the runtime, trying to read content of circular symlink would throw ELOOP error.